### ISSUE TITLE
infinite grad norm using PL 1.4.0

### ISSUE THREAD

**Issue by richarddwang** [PyTorchLightning/pytorch-lightning#8689]
## 🐛 Bug

Training the same code with the same seed with PL 1.4.0 / 1.3.8,
only when using PL 1.4.0,
after a few thousands steps, I got Pytorch warns me grad_norm is infinite, and loss barely decrease.
But with PL 1.3.8 all works well,
there is no infinite grad_norm warning and loss decrease quickly.

### Other possibley relevant details
gradient_clip_val=1.0
precision=16

### Environment

 - PyTorch Lightning Version (e.g., 1.3.0): 1.4.0/1.3.8
 - PyTorch Version (e.g., 1.8): 1.9.0
 - Python version: 3.9.5
 - OS (e.g., Linux): Ubuntu
 - CUDA/cuDNN version: 11.2
 - How you installed PyTorch (`conda`, `pip`, source): conda

**Comment by Borda** [comment]
@richarddwang mind share an example, ideally Colab?

**Comment by richarddwang** [comment]
I am currently doing pretraining language model with large data, and the repo is private. I am sorry that I may not be able to provide any example. But since I am happy with PL 1.3.8 now, you may just keep this issue stale and come back after you find new clues elsewhere (?).
